airbyte-commons/src/main/resources/log4j2.xml

```diff
@@ -12,8 +12,6 @@
 
         <Property name="s3-bucket">${sys:S3_LOG_BUCKET:-${env:S3_LOG_BUCKET}}</Property>
         <Property name="s3-region">${sys:S3_LOG_BUCKET_REGION:-${env:S3_LOG_BUCKET_REGION}}</Property>
-        <Property name="s3-aws-key">${sys:AWS_ACCESS_KEY_ID:-${env:AWS_ACCESS_KEY_ID}}</Property>
```

Comment:
@msaffitz can you explain why the removal of these lines are necessary? What happens if you introduce the `sts` library without removing these?

Reply:
I'm not actually sure; I had the same question above (https://github.com/airbytehq/airbyte-platform/pull/263#issuecomment-1657240363) and @bothra90 indicated that removing them was necessary to get things functioning. @bothra90 -- do you have more context on this?

Reply:
Yes, I was unable to create a source from the console and seeing the following error log in `airbyte-server` process:
```
Cannot end publish with com.van.logging.aws.S3PublishHelper@53f4b9a9 due to error: Cannot end publishing: Cannot publish to S3: The AWS Access Key Id you provided does not exist in our records. (Service: Amazon S3; Status Code: 403; Error Code: InvalidAccessKeyId; Request ID: TY5JJZ76SMD9BA86; S3 Extended Request ID: WEK4bCkgavHUqbgoywsSNVMwBt14/ke89r9mC/usS879oBQ1MYNw+S/2urb3xedmfWxWBpfj90TMLqnYL+hCfw==; Proxy: null)
```

Reply:
I discussed this internally with some folks that have more familiarity around our Log4J2 setup. Simply removing these properties won't work -- we use an underlying [log4j-appender implementation](https://github.com/bluedenim/log4j-s3-search) that will need to support IRSA. 

We opened [an issue](https://github.com/bluedenim/log4j-s3-search/issues/140) for this in that project so we'll wait and see what the maintainer says.

Reply:
FWIW, it did the job for me. Happy to jump on a quick Slack huddle and demo it to you.

Reply:
It appears that the underlying appender library that I linked above actually supports roles by default. The `s3-aws-key` and `s3-aws-secret` properties are optional, for situations where role-based credentials aren't available. That said, I'm concerned that if we simply delete these properties from the xml configuration, we could break Airbyte installations that are using those properties instead of roles. 

An ideal solution would be backwards compatible for such installations, though I'm not sure what that would look like, since it seems like as soon as these <Property> tags are defined, that appender will try to use them, even if the value is blank

Reply:
I understand the concern. Ultimately though, the passed parameters get used [here](https://github.com/bluedenim/log4j-s3-search/blob/90317ec6efbd666d7230d9fdcac265921e044171/appender-core/src/main/java/com/van/logging/aws/AwsClientHelpers.java#L11-L39) and if the `s3AwsSecret` and `s3AwsKey` are not provided, the library will use the default AWS providers chain which will correctly use `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`. This is how these variables in the configuration are populated in the first place, so I don't suspect this would be an issue.

Having said that, I can try to test this out on my end and make sure nothing breaks.

Reply:
@bothra90 @weichunnn thanks for pointing this out, and apologies for the slow-moving back and forth here. I agree that it seems like these XML properties look unnecessary. @bothra90 if you're able to test and verify that this works as expected on your end, I'd love to see your results. 

Otherwise, I'll open this PR against our internal repo where I can run our full suite of tests. Thanks for bearing with me here!

Reply:
Thanks @pmossman! Any updates on the review internally? 👀